tests/testsuite/corrupt_git.rs

```diff
@@ -0,0 +1,165 @@
+use std::fs;
+use std::path::{Path, PathBuf};
+
+use cargotest::support::paths;
+use cargotest::support::{git, project, execs};
+use hamcrest::assert_that;
+
+#[test]
+fn deleting_database_files() {
+    let project = project("foo");
+    let git_project = git::new("bar", |project| {
+        project
+            .file("Cargo.toml", r#"
+                [project]
+                name = "bar"
+                version = "0.5.0"
+                authors = []
+            "#)
+            .file("src/lib.rs", "")
+    }).unwrap();
+
+    let project = project
+        .file("Cargo.toml", &format!(r#"
+            [project]
+            name = "foo"
+            version = "0.5.0"
+            authors = []
+
+            [dependencies]
+            bar = {{ git = '{}' }}
+        "#, git_project.url()))
+        .file("src/lib.rs", "")
+        .build();
+
+    assert_that(project.cargo("build"), execs().with_status(0));
+
+    let mut files = Vec::new();
+    find_files(&paths::home().join(".cargo/git/db"), &mut files);
+    assert!(files.len() > 0);
+
+    let log = "cargo::sources::git=trace";
+    for file in files {
+        if !file.exists() {
+            continue
+        }
+        println!("deleting {}", file.display());
+        fs::remove_file(&file).unwrap();
```

Comment:
Ouuuuu, I **love** this approach to testing! :100: :+1: :muscle: 

Let's move `find_files` fn definition to the bottom of the file: it is currently *between* the two tests that use it :)